js/reducers/passageReducer.js

```diff
@@ -100,4 +101,14 @@ function passageReducer(state = initialState, action) {
   }
 }
 
-export default passageReducer;
+function stateSavingPassageReducer(state, action) {
+  if (!state) {
+    state = assignToEmpty(defaultState, restoreState());
```

Comment:
For functional purity, the state to restore to needs to be captured in the arguments to the reducer. I.e., the reducer can't read anything from the outside world. So I've implemented this as a `RESTORE_STATE` action that has a `state` property representing the state to restore to. This action will get fired at load time in app.js.